sdks/java/io/fileschematransform/src/main/java/org/apache/beam/sdk/io/fileschematransform/FileWriteSchemaTransformConfiguration.java

```diff
@@ -0,0 +1,215 @@
+  @AutoValue
+  public abstract static class CsvConfiguration {
+
+    /** Content written to the top of every sharded file prior to the header. */
+    @Nullable
+    public abstract String getPreamble();
+
+    /**
+     * The format of the written CSV file. See <a
+     * href="https://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVFormat.html">org.apache.commons.csv.CSVFormat</a>
+     * for allowed values, stringified in lowercase. Defaults to <a
+     * href="https://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVFormat.html#DEFAULT">CSVFormat.DEFAULT</a>
+     */
+    @Nullable
+    public abstract String getCsvFormat();
+
+    @AutoValue.Builder
+    public abstract static class Builder {
+
+      /** Content written to the top of every sharded file prior to the header. */
+      public abstract Builder setPreamble(String value);
+
+      /**
+       * The format of the written CSV file. See <a
+       * href="https://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVFormat.html">org.apache.commons.csv.CSVFormat</a>
+       * for allowed values, stringified in lowercase. Defaults to <a
+       * href="https://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVFormat.html#DEFAULT">CSVFormat.DEFAULT</a>
+       */
+      public abstract Builder setCsvFormat(String value);
+
+      public abstract CsvConfiguration build();
+    }
+  }
+
+  /** Configures extra details related to writing Parquet formatted files. */
+  @DefaultSchema(AutoValueSchema.class)
+  @AutoValue
+  public abstract static class ParquetConfiguration {
+
+    /**
+     * Specifies compression codec. See org.apache.parquet.hadoop.metadata.CompressionCodecName for
+     * allowed names, stringified in lowercase format.
+     */
+    public abstract String getCompressionCodecName();
+
+    /** Specify row-group size; if not set or zero, a default is used by the underlying writer. */
+    public abstract Integer getRowGroupSize();
+
+    @AutoValue.Builder
+    public abstract static class Builder {
+
+      /**
+       * Specifies compression codec. See org.apache.parquet.hadoop.metadata.CompressionCodecName
+       * for allowed names, stringified in lowercase format.
+       */
+      public abstract Builder setCompressionCodecName(String value);
+
+      /** Specify row-group size; if not set or zero, a default is used by the underlying writer. */
+      public abstract Builder setRowGroupSize(Integer value);
+
+      public abstract ParquetConfiguration build();
+    }
+  }
+
+  /** Configures extra details related to writing XML formatted files. */
+  @DefaultSchema(AutoValueSchema.class)
+  @AutoValue
+  public abstract static class XmlConfiguration {
+
+    /** Sets the enclosing root element for the generated XML files. */
+    public abstract String getRootElement();
+
+    /** Sets the charset used to write the file. */
+    public abstract String getCharset();
```

Comment:
I suppose default here is utf8?

Reply:
Good call.  Robert caught this as well.  And yes :-)

Reply:
cool thanks Damon!